main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -3080,11 +3071,20 @@ object Weeder2 {
     val trailingDot = tryPick(TreeKind.TrailingDot, tree).nonEmpty
     assert(idents.nonEmpty) // We require at least one element to construct a qname
     val first = idents.head
-    val ident = idents.last
-    val nnameIdents = idents.dropRight(1)
-    val loc = SourceLocation(isReal = true, first.loc.sp1, ident.loc.sp2)
-    val nname = Name.NName(nnameIdents, loc)
-    Name.QName(nname, ident, loc, trailingDot)
+    val last = idents.last
+    val loc = SourceLocation(isReal = true, first.loc.sp1, last.loc.sp2)
+
+    // If there is a trailing dot, we use all the idents as namespace and use "" as the ident
+    if (trailingDot) {
+      val nname = Name.NName(idents, loc)
```

Comment:
Can we add an example here in a comment. To explain how the generated QName looks like.